specs/merge/beacon-chain.md

```diff
@@ -156,54 +156,52 @@ def process_block(state: BeaconState, block: BeaconBlock) -> None:
+Let `get_execution_state(execution_state_root: Bytes32) -> ExecutionState`  be the function that given the root hash returns a copy of Ethereum execution state. 
 The body of the function is implementation dependent.
 
-##### `application_state_transition`
+##### `execution_state_transition`
 
-Let `application_state_transition(application_state: ApplicationState, application_payload: ApplicationPayload) -> None` be the transition function of ethereum application state. 
+Let `execution_state_transition(execution_state: ExecutionState, execution_payload: ExecutionPayload) -> None` be the transition function of Ethereum execution state. 
 The body of the function is implementation dependent.
 
-*Note*: `application_state_transition` must throw `AssertionError` if either the transition itself or one of the post-transition verifications has failed.
+*Note*: `execution_state_transition` must throw `AssertionError` if either the transition itself or one of the pre or post conditions has failed.
 
-##### `process_application_payload`
+##### `process_execution_payload`
 
 ```python
-def process_application_payload(state: BeaconState, body: BeaconBlockBody) -> None:
+def process_execution_payload(state: BeaconState, body: BeaconBlockBody) -> None:
     """
     Note: This function is designed to be able to be run in parallel with the other `process_block` sub-functions
     """
 
-    if not is_transition_completed(state):
-        assert body.application_payload == ApplicationPayload()
-        return
-
-    if not is_transition_block(state, body):
-        assert body.application_payload.parent_hash == state.latest_application_block_header.block_hash
-        assert body.application_payload.number == state.latest_application_block_header.number + 1
-
-    application_state = get_application_state(state.latest_application_block_header.state_root)
-    application_state_transition(application_state, body.application_payload)
-
-    state.latest_application_block_header = ApplicationBlockHeader(
-        block_hash=application_payload.block_hash,
-        parent_hash=application_payload.parent_hash,
-        coinbase=application_payload.coinbase,
-        state_root=application_payload.state_root,
-        number=application_payload.number,
-        gas_limit=application_payload.gas_limit,
-        gas_used=application_payload.gas_used,
-        receipt_root=application_payload.receipt_root,
-        logs_bloom=application_payload.logs_bloom,
-        transactions_root=hash_tree_root(application_payload.transactions),
+    execution_payload = body.execution_payload
+
+    if is_transition_completed(state):
+        assert execution_payload.parent_hash == state.latest_execution_payload_header.block_hash
+        assert execution_payload.number == state.latest_execution_payload_header.number + 1
+
+    execution_state = get_execution_state(state.latest_execution_payload_header.state_root)
```

Comment:
Don't we need to skip this execution state transition if we are before the merge transition?

Maybe a blocking condition at the top?

Reply:
Definitely, I have completely missed this code branch. Let me fix